signing/multisig/blsMultisig_test.go

```diff
@@ -290,232 +333,447 @@ func TestBLSMultiSigner_CreateOK(t *testing.T) {
 
-	verifErr := multiSig.VerifySignatureShare(ownIndex, sigShare, msg, []byte(""))
+	t.Run("with rogue key prevention", func(t *testing.T) {
+		sigShare, multiSig := createSignerAndSigShareBLS(pubKeys, privKey, kg, ownIndex, msg, llSigner)
+
+		verifErr := multiSig.VerifySignatureShare(ownIndex, sigShare, msg, []byte(""))
+
+		assert.Nil(t, verifErr)
+	})
+	t.Run("with KOSK", func(t *testing.T) {
+		sigShare, multiSig := createSignerAndSigShareBLS(pubKeys, privKey, kg, ownIndex, msg, llSignerKOSK)
 
-	assert.Nil(t, verifErr)
+		verifErr := multiSig.VerifySignatureShare(ownIndex, sigShare, msg, []byte(""))
+
+		assert.Nil(t, verifErr)
+	})
 }
 
 func TestBLSMultiSigner_AddSignatureShareNilSigShouldErr(t *testing.T) {
 	t.Parallel()
 
 	ownIndex := uint16(3)
-	privKey, _, pubKeys, kg, llSigner := generateMultiSigParamsBLS(4, ownIndex)
-	multiSig, _ := multisig.NewBLSMultisig(llSigner, pubKeys, privKey, kg, ownIndex)
+	hasher := &mock.HasherSpongeMock{}
+	llSigner := &llsig.BlsMultiSigner{Hasher: hasher}
+	llSignerKOSK := &llsig.BlsMultiSignerKOSK{}
+	privKey, _, pubKeys, kg := generateMultiSigParamsBLS(4, ownIndex)
+
+	t.Run("with rogue key prevention", func(t *testing.T) {
+		multiSig, _ := multisig.NewBLSMultisig(llSigner, pubKeys, privKey, kg, ownIndex)
+
+		err := multiSig.StoreSignatureShare(ownIndex, nil)
+
+		assert.Equal(t, crypto.ErrNilSignature, err)
+	})
+	t.Run("with KOSK", func(t *testing.T) {
+		multiSig, _ := multisig.NewBLSMultisig(llSignerKOSK, pubKeys, privKey, kg, ownIndex)
 
-	err := multiSig.StoreSignatureShare(ownIndex, nil)
+		err := multiSig.StoreSignatureShare(ownIndex, nil)
 
-	assert.Equal(t, crypto.ErrNilSignature, err)
+		assert.Equal(t, crypto.ErrNilSignature, err)
+	})
 }
 
 func TestBLSMultiSigner_AddSignatureShareIndexOutOfBoundsIndexShouldErr(t *testing.T) {
 	t.Parallel()
 
 	ownIndex := uint16(3)
-	privKey, _, pubKeys, kg, llSigner := generateMultiSigParamsBLS(4, ownIndex)
-	multiSig, _ := multisig.NewBLSMultisig(llSigner, pubKeys, privKey, kg, ownIndex)
+	hasher := &mock.HasherSpongeMock{}
+	llSigner := &llsig.BlsMultiSigner{Hasher: hasher}
+	llSignerKOSK := &llsig.BlsMultiSignerKOSK{}
+	privKey, _, pubKeys, kg := generateMultiSigParamsBLS(4, ownIndex)
 
-	sigShare, _ := multiSig.CreateSignatureShare([]byte("message"), []byte(""))
+	t.Run("with rogue key prevention", func(t *testing.T) {
+		multiSig, _ := multisig.NewBLSMultisig(llSigner, pubKeys, privKey, kg, ownIndex)
+		sigShare, _ := multiSig.CreateSignatureShare([]byte("message"), []byte(""))
 
-	err := multiSig.StoreSignatureShare(15, sigShare)
+		err := multiSig.StoreSignatureShare(15, sigShare)
 
-	assert.Equal(t, crypto.ErrIndexOutOfBounds, err)
+		assert.Equal(t, crypto.ErrIndexOutOfBounds, err)
+	})
+	t.Run("with KOSK", func(t *testing.T) {
+		multiSig, _ := multisig.NewBLSMultisig(llSignerKOSK, pubKeys, privKey, kg, ownIndex)
+		sigShare, _ := multiSig.CreateSignatureShare([]byte("message"), []byte(""))
```

Comment:
can't we use `createSignerAndSigShareBLS` here also?

Reply:
done